serlcd/serlcd.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// This package provides an implementation for the SparkFun SerLCD intelligent
+// LCD display. This display provides hardware interfaces for SPI, I2C, and
+// UART. Implements conn.display.TextDisplay
+package serlcd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"periph.io/x/conn/v3"
+	"periph.io/x/conn/v3/display"
+)
+
+// Representation of a SerLCD display.
+type Dev struct {
+	conn conn.Conn
+	w    io.Writer
+	cols int
+	rows int
+	// Display on/off, Curosr, Blink
+	displayDCB byte
+}
+
+const (
+	DefaultI2CAddress uint16 = 0x72
+)
+
+var settingMode byte = 0x7c
+var cmdMode byte = 0xfe
+var clear = []byte{settingMode, 0x2d}
+
+func wrap(err error) error {
+	return fmt.Errorf("serlcd: %w", err)
+}
+
+// Create a SerLCD display using a hardware interface that provides io.Writer.
+// That can be the i2c.Bus, or a 3rd Party UART library for serial
+// communications.
+func NewSerLCD(writer io.Writer, rows, cols int) *Dev {
```

Comment:
Use `New`, as `serlcd.NewSerLCD` stutters. Maybe `NewSerial`?

Reply:
The proper name of the product is SerLCD